Tests/Shared/Core/UnitTestBase.swift

```diff
@@ -0,0 +1,115 @@
+@testable import CioInternalCommon
+import Foundation
+import XCTest
+
+/// Serves as the base class for all tests within the SDK, offering common setup, teardown, and
+/// utility methods shared across test cases. This class simplifies the initialization of the SDK/modules,
+/// mock objects, test data, and other shared resources, ensuring a consistent testing environment and
+/// minimizing boilerplate in individual test cases.
+///
+/// The generic `Component` parameter represents the module interface, providing convenient and direct access
+/// to the module's public APIs being tested. `Component` can be specified as the SDK (e.g., `CustomerIO`)
+/// or a specific module (e.g., `MessagingPush`).
+///
+/// For SDK-wide tests, child classes can conveniently inherit from `UnitTest`, designed specifically for testing only SDK APIs.
+open class UnitTestBase<Component>: XCTestCase {
+    public let testWriteKey = "test"
+    public var diGraphShared: DIGraphShared!
+    public var log: Logger { diGraphShared.logger }
+    public var globalDataStore: GlobalDataStore { diGraphShared.globalDataStore }
+
+    public let testSiteId = "testing"
+    public var diGraph: DIGraph!
+    public var sdkConfig: SdkConfig { diGraph.sdkConfig }
+
+    public var jsonAdapter: JsonAdapter { JsonAdapter(log: log) }
+    public var lockManager: LockManager { LockManager() }
+    public var dateUtilStub: DateUtilStub!
+    public var threadUtilStub: ThreadUtilStub!
+
+    override open func setUp() {
+        setUp(enableLogs: false, modifySdkConfig: nil)
+    }
+
+    /**
+     Perform setup before each test function runs in your test class.
+     Example:
+     ```
+     override func setUp() {
+       super.setUp() // <-- this line calls this function in UnitTest file.
+
+       // do some setup logic, here.
+     }
+     ```
+
+     @param enableLogs Enables logging for the test class. Can be useful for debugging. Disabled by default it's too noisey and unhelpful when logs are enabled for all tests.
+     @param modifySdkConfig Allows you to change configuration options before the SDKConfig instnace is created for you.
+     */
+    open func setUp(
+        enableLogs: Bool = false,
+        siteId: String? = nil,
+        modifySdkConfig: ((inout SdkConfig) -> Void)?
+    ) {
+        var newSdkConfig = SdkConfig.Factory.create(siteId: siteId ?? testSiteId, apiKey: "", region: Region.US)
+        if enableLogs {
+            newSdkConfig.logLevel = CioLogLevel.debug
+        }
+        modifySdkConfig?(&newSdkConfig)
+
+        diGraphShared = DIGraphShared()
+        diGraph = DIGraph(sdkConfig: newSdkConfig)
+
+        // setup and override dependencies before creating SDK instance, as Shared graph may be initialized and used immediately
+        setUpDependencies()
+        // setup SDK instance and set necessary components for testing
+        initializeSDKComponents()
```

Comment:
Used open methods for setting up SDK/module so they can be overridden and implemented by child classes as needed